src/app/loadout-drawer/subclass-drawer/SubclassPlugDrawer.tsx

```diff
@@ -0,0 +1,186 @@
+    const initiallySelected = Object.values(socketOverrides)
+      .map((hash) => defs?.InventoryItem.get(hash))
+      .filter(isPluggableItem);
+    const { plugs, aspects, fragments } = getPlugsForSubclass(defs, profileResponse, subclass);
+    return {
+      initiallySelected,
+      plugs: Array.from(plugs),
+      aspects,
+      fragments,
+    };
+  }, [defs, profileResponse, socketOverrides, subclass]);
+
+  const onAcceptInternal = useCallback(
+    (selected: PluggableInventoryItemDefinition[]) => {
+      if (!subclass.sockets) {
+        return;
+      }
+
+      const remainingPlugs = Array.from(selected);
+      const newOverrides: SocketOverrides = {};
+
+      for (const socket of subclass.sockets.allSockets) {
+        const socketPlugsetHashes = getPlugHashesForSocket(socket, profileResponse);
+        for (const [index, plug] of remainingPlugs.entries()) {
+          if (socketPlugsetHashes.some((hash) => hash === plug.hash)) {
+            newOverrides[socket.socketIndex] = plug.hash;
+            remainingPlugs.splice(index, 1);
+            break;
+          }
+        }
+      }
+      onAccept(newOverrides);
+    },
+    [onAccept, profileResponse, subclass.sockets]
+  );
+
+  // Determines whether an ability, aspect or fragment is currently selectable
+  // - Any: only a single instace can be selected at a time
+  // - Ability: Only a single ability can be selected at a time.
+  // - Fragments: the energy level of the aspects determines the number that can be selected
+  // - Aspects: A maximum of 2 can be selected.
+  const isPlugSelectable = useCallback(
+    (plug: PluggableInventoryItemDefinition, selected: PluggableInventoryItemDefinition[]) => {
+      if (selected.some((s) => s.hash === plug.hash)) {
+        return false;
+      }
+
+      // Abilities handling
+      if (!aspects.has(plug) && !fragments.has(plug)) {
+        // TODO (ryan) this is annoying as you cannot just click a different ability to auto
+        // deselect the current one. We should come up with a way to allow swapping by a single
+        // click
+        return !selected.some((s) => s.plug.plugCategoryHash === plug.plug.plugCategoryHash);
+      }
+
+      // Aspects handling
+      const selectedAspects = selected.filter((plugDef) => aspects.has(plugDef));
+      if (aspects.has(plug)) {
+        return selectedAspects.length < MAX_ASPECTS;
+      }
+
+      // Fragments handling
+      const selectedFragments = selected.filter((plugDef) => fragments.has(plugDef));
+      const allowedFragments = _.sumBy(
+        selectedAspects,
+        (aspect) => aspect.plug.energyCapacity?.capacityValue || 0
+      );
+      if (fragments.has(plug)) {
+        return selectedFragments.length < allowedFragments;
+      }
+
+      return true;
+    },
+    [aspects, fragments]
+  );
+
+  return (
+    <PlugDrawer
+      title={t('Loadouts.AspectsAndFragments')}
+      searchPlaceholder={t('Loadouts.SearchAspectsAndFragments')}
```

Comment:
i think these i18n keys are missing

Reply:
Yeah they are, I will add them in my final cleanup